### PR TITLE
Chat page

### PR DESCRIPTION
**Pull request by jacklansing**
Implementation feels a bit messy right now, but I think we'll be able to clean a lot of this up as it comes together. Feel free to let me know if any changes requested, and I'll take a look this weekend. 

Thanks!

### CODE REVIEW

**Comment by jacklansing** [comment]
Note: I think I largely ignored the CSS since we're not very committed to it, but Andrew's last PR included some global styles in the Login.css that collided with everyone else's. I'll leave it up to you to decide if they should be temporarily changed or we want to leave as-is for now.

**Comment by malcolmkiano** [comment]
> Note: I think I largely ignored the CSS since we're not very committed to it, but Andrew's last PR included some global styles in the Login.css that collided with everyone else's. I'll leave it up to you all to decide if they should be temporarily changed or we want to leave as-is for now.

I'll ask him to scope his styles by adding a class name before all his selectors. Makes it easier for everyone to do their own thing.

File: src/components/Chat/Chat.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        Authorization: `Bearer ${window.localStorage.getItem(config.TOKEN_KEY)}`
+      }
+    })
+      .then(res => res.json())
+      .then(chats => this.setState({ ...chats }))
+      .catch(error => this.setState({ error }));
+  };
+
+  getNameInitials(firstName, lastName) {
+    // TODO: Refactor into shared util with ChatMessages
```
Comment (malcolmkiano) [inline src/components/Chat/Chat.js:42]: Appreciate the doc

File: src/components/ChatMessageForm/ChatMessageForm.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    const { body } = this.state;
+    const { recipient_id, project_id } = this.props;
+    const newMessage = {
+      body,
+      recipient_id,
+      project_id
+    };
+
+    fetch(`${config.API_ENDPOINT}/chats`, {
```
Comment (malcolmkiano) [inline src/components/ChatMessageForm/ChatMessageForm.js:32]: These are gonna move into the util, yes?

File: src/components/ChatMessages/ChatMessages.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      first_name: '',
+      last_name: '',
+      project_name: '',
+      error: null,
+      interval_id: null
+    };
+  }
+
+  componentDidMount() {
+    /* If user arrives at this route with a chat_id, send them back
```
Comment (malcolmkiano) [inline src/components/ChatMessages/ChatMessages.js:23]: You mean without a chat id, correct? No big deal, just wondering

File: src/components/ChatMessageForm/ChatMessageForm.js (hunk 2 above)
Comment (jacklansing) [reply src/components/ChatMessageForm/ChatMessageForm.js:32]: They definitely could, especially if they are going to be used elsewhere at all.

File: src/components/ChatMessages/ChatMessages.js (hunk 3 above)
Comment (jacklansing) [reply src/components/ChatMessages/ChatMessages.js:23]: Ah yeah that's a typo, will be 'arrives at this route without a chat_id'

**Pull request merged**